### ISSUE TITLE
Setup/installation failure

### ISSUE THREAD

**Issue by cmt29** [papoteur-mga/elograf#9]
I've just tried downloading and installing the latest version but am running into an installation error that I don't remember getting previously. I haven't moved nerd-dictation from where I had it before (~/bin). I've tried moving it to ~/.local/bin but still doesn't work. Both paths are in my path and nerd-dictation runs fine independently. Any thoughts?

Here's the output from `sudo python3 setup.py install`:

```
Searching for nerd-dictation
Reading https://pypi.org/simple/nerd-dictation/
Couldn't find index page for 'nerd-dictation' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.org/simple/
No local packages or working download links found for nerd-dictation
error: Could not find suitable distribution for Requirement.parse('nerd-dictation')
```

**Comment by cmt29** [comment]
OK thanks. Now I'm getting this error upon running elograf:
```
Traceback (most recent call last):
  File "/usr/bin/elograf", line 4, in <module>
    __import__('pkg_resources').run_script('EloGraf==0.4.0', 'elograf')
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 661, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1432, in run_script
    .format(**locals()),
pkg_resources.ResolutionError: Script 'scripts/elograf' not found in metadata at '/home/chris/bin/elograf-master/EloGraf.egg-info'
```

**Comment by papoteur-mga** [comment]
Hello,
I don't reproduce this error.

**Comment by cmt29** [comment]
OK, I've reverted to v3.2 for now.
Can you give me any guidance on what might be causing the above error?

**Comment by papoteur-mga** [comment]
I don't know how this happens.
When I execute the `sudo python3 setup.py install`, the installation go in /usr/local/. However, it seems that for you the installation is in /usr for the script elograf.
How did you do the installation?
Know also that Python 3.6 is "end of life". You will need this patch. https://github.com/papoteur-mga/elograf/issues/5#issuecomment-1067006506
https://peps.python.org/pep-0494/#lifespan

**Comment by jaygotango** [comment]
Same error trying to install, commenting out the nerd-dictation requirement gives me this result:

Traceback (most recent call last):
  File "/usr/local/bin/elograf", line 4, in <module>
    __import__('pkg_resources').run_script('EloGraf==0.4.0', 'elograf')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 667, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1463, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python3.8/dist-packages/EloGraf-0.4.0-py3.8.egg/EGG-INFO/scripts/elograf", line 3, in <module>
    import eloGraf.elograf as elograf
  File "/usr/local/lib/python3.8/dist-packages/EloGraf-0.4.0-py3.8.egg/eloGraf/elograf.py", line 13, in <module>
    import ujson
ModuleNotFoundError: No module named 'ujson'


That was resolved by installing the ujson package (python3-ujson on 20.04)

**Comment by papoteur-mga** [comment]
Thanks for the report.
In the configuration, did you choose a model, either by downloading or by adding one already present?
The Exit not working is surprising. I didn't see that. Is there something reported in console?

**Comment by papoteur-mga** [comment]
I have pushed a fix to avoid the loop.
The exit didn't work because the background program managing it is stopped. You will have a zombie icon until the next boot/connection.

**Comment by jaygotango** [comment]
> Thanks for the report. In the configuration, did you choose a model, either by downloading or by adding one already present? The Exit not working is surprising. I didn't see that. Is there something reported in console?

Model was already installed in ~/.config/nerd-dictation/model

nerd-dictation itself is in the $PATH (in ~/.local/bin)

**Comment by jaygotango** [comment]
> I have pushed a fix to avoid the loop. The exit didn't work because the background program managing it is stopped. You will have a zombie icon until the next boot/connection.

Just installed it (after commenting out the nerd-dictation requirement in setup.cfg) and it works!

The only problem is that whenever you activate it, you get the Configuration screen and must click Save then Cancel to get it to work.

**Comment by papoteur-mga** [comment]
Yes, I can still improve the UI.

**Comment by Lume6** [comment]
Hi,
Thank you for this program that will make nerd-dictation easier to use. The installation went well but the launch of elograf fails:

> ~/apps/elograf-master$ elograf
Traceback (most recent call last):
  File "/usr/local/bin/elograf", line 4, in <module>
    __import__('pkg_resources').run_script('EloGraf==0.4.0', 'elograf')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3254, in <module>
    def _initialize_master_working_set():
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3237, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3266, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'nerd-dictation' distribution was not found and is required by EloGraf

I think it's due to the absence of nerd-dictation which is in ~/nerd-dictation with a symbolic link to /usr/local. 

 Other information:

-  elograf is installed in  /usr/local/bin
-  Model was already installed in ~/.config/nerd-dictation/model
-  nerd-dictation is installed in ~/.config/nerd-dictation
-  nerd-dictation is installed in ~/.local/bin/nerd-dictation

I don't know what path to point to in order for Elograf to detect nerd-dictation.
Thank you for your help !

**Comment by Lume6** [comment]
> 

I forgot to tell you I commented the lines 22-23 in file setup.cfg.

**Comment by papoteur-mga** [comment]
I have no released the 0.4.1, thus nerd-dictation is no more required as module.

**Comment by papoteur-mga** [comment]
There is now 0.5.3.
Is this release OK?
